### PR TITLE
Fix a race in a test of the RecoveringIndexer

### PR DESCRIPTION
**Pull request by mziolekda**
Make sure the list of openSubscriptions is updated before the resulting future is passed to the test code for further processing.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala [hunk 1]
```diff
@@ -367,11 +367,10 @@ object RecoveringIndexerSpec {
             }
           })
         })(handle => {
-          val complete = handle.stop()
-          complete.onComplete { _ =>
+          handle.stop().transform { complete =>
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala:370]: What is the difference? (Except for the nicer construction)

Comment (kamil-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala:370]: I think previously we did:
```
val complete = handle.stop()
complete.onComplete { doStuff() }
complete // return complete immediately without waiting for doStuff()
```

and now it is:
```
handle.stop().transform { complete =>
doStuff() // do stuff when stop() completes
complete // return complete but this time only when doStuff() is completed
}

Comment (tudor-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala:370]: I see, thanks

Comment (rautenrieth-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala:370]: I think: Before, the `.onComplete` handler and the `.map` transformation could run in any order. With this change, it is guaranteed that the subscription is removed first.

**Review by tudor-da — APPROVED** [review]

File: ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala (hunk 1 above)
Comment (mziolekda) [reply ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala:370]: In the past the lambda in onComplete was racing against the client code attached to the returned future. Now the lambda is executed strictly before.

**Pull request merged**